php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -55,6 +91,229 @@ public ExpressionTree expression(InternalSyntaxToken token) {
+        }
+      }
+    }
+    return new ListExpressionTreeImpl(listToken, openParenthesis, new SeparatedList(expressions, commas), closeParenthesis);
+  }
+
+  public AssignmentExpressionTree listExpressionAssignment(ExpressionTree listExpression, InternalSyntaxToken equalToken, ExpressionTree expression) {
+    return new AssignmentExpressionTreeImpl(Kind.ASSIGNMENT, listExpression, equalToken, expression);
+  }
+
+  public ComputedVariableTree computedVariableName(InternalSyntaxToken openCurly, ExpressionTree expression, InternalSyntaxToken closeCurly) {
+    return new ComputedVariableTreeImpl(openCurly, expression, closeCurly);
+  }
+
+  public VariableIdentifierTree variableIdentifier(InternalSyntaxToken variableIdentifier) {
+    return new VariableIdentifierTreeImpl(new IdentifierTreeImpl(variableIdentifier));
+  }
+
+  public CompoundVariableTree compoundVariable(InternalSyntaxToken openDollarCurly, ExpressionTree expression, InternalSyntaxToken closeDollarCurly) {
+    return new CompoundVariableTreeImpl(openDollarCurly, expression, closeDollarCurly);
+  }
+
+  public ArrayAccessTree dimensionalOffset(InternalSyntaxToken openCurly, Optional<ExpressionTree> expression, InternalSyntaxToken closeCurly) {
+    if (expression.isPresent()) {
+      return new ArrayAccessTreeImpl(openCurly, expression.get(), closeCurly);
+    }
+    return new ArrayAccessTreeImpl(openCurly, closeCurly);
+  }
+
+  public ExpressionTree variableWithoutObjects(Optional<List<InternalSyntaxToken>> dollars, VariableTree compoundVariable, Optional<List<ArrayAccessTree>> offsets) {
+    ExpressionTree result = compoundVariable;
+    for (ExpressionTree partialArrayAccess : optionalList(offsets)) {
+      result = ((ArrayAccessTreeImpl) partialArrayAccess).complete(result);
+    }
+
+    if (dollars.isPresent()) {
+      result = new VariableVariableTreeImpl(dollars.get(), result);
+    }
+
+    return result;
+  }
+
+  public ArrayAccessTree alternativeDimensionalOffset(InternalSyntaxToken openBrace, Optional<ExpressionTree> offset, InternalSyntaxToken closeBrace) {
+    if (offset.isPresent()) {
+      return new ArrayAccessTreeImpl(openBrace, offset.get(), closeBrace);
+    }
+    return new ArrayAccessTreeImpl(openBrace, closeBrace);
+  }
+
+  public IdentifierTree newStaticIdentifier(InternalSyntaxToken staticToken) {
+    return new IdentifierTreeImpl(staticToken);
+  }
+
+  public ReferenceVariableTree referenceVariable(InternalSyntaxToken ampersand, ExpressionTree variable) {
+    return new ReferenceVariableTreeImpl(ampersand, variable);
+  }
+
+  public SpreadArgumentTree spreadArgument(InternalSyntaxToken ellipsis, ExpressionTree expression) {
+    return new SpreadArgumentTreeImpl(ellipsis, expression);
+  }
+
+  public FunctionCallTree functionCallParameterList(InternalSyntaxToken openParenthesis, Optional<Tuple<ExpressionTree, Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>>>> arguments, InternalSyntaxToken closeParenthesis) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 231 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)